src/main/java/Duke.java

```diff
@@ -1,10 +1,203 @@
+            Task task = tasks.get(taskNumber - 1);
+            task.unmarkAsDone();
+
+            printMarkNotDone();
+            printTask(task);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("The given task number does not exist. ):");
+            //throw in here
+        } catch (NumberFormatException e) {
+            System.out.println("The task index must be numeric.");
+        } catch (DukeException e) {
+            System.out.println("The task is already marked as not done.");
+        }
+    }
+    public static void addTodo(ArrayList<Task> tasks, String description) {
+        printAddTask();
+
+        Task task = new Todo(description);
+        tasks.add(task);
+
+        printTask(task);
+        printNoOfTasks(tasks.size());
+    }
+
+    public static void addDeadline(ArrayList<Task> tasks, String command) /*throws DukeException*/ {
+
+        if (command.contains("/by")) {
+            String[] components = command.split(" /by");
+
+            try {
+                Task task = new Deadline(components[0], components[1]);
+                tasks.add(task);
+                printAddTask();
+                printTask(task);
+                printNoOfTasks(tasks.size());
+            } catch (ArrayIndexOutOfBoundsException e) {
+                System.out.println("Please specify a time for the deadline.");
+            }
+
+        } else {
+            System.out.println("Invalid format. Remember to use '/by' to indicate the time.");
+            //throw new DukeException();
+        }
+    }
+
+    public static void addEvent(ArrayList<Task> tasks, String command) {
+
+        if (command.matches("(.*)" + "/from" + "(.*)" + "/to" + "(.*)")) {
+
+            try {
+                String[] components = command.split(" /from | /to "); //split string using "/from" and "/to"
+                Task task = new Event(components[0], components[1], components[2]);
+                tasks.add(task);
+
+                printAddTask();
+                printTask(task);
+                printNoOfTasks(tasks.size());
+            } catch (ArrayIndexOutOfBoundsException e){
+                System.out.println("Please specify both the starting and ending time of the event");
+            }
+        } else {
+            System.out.println("Incorrect format. Specify events in the format 'event A /from B to /C'");
+        }
+
+
+    }
+    public static void printAllTasks(ArrayList<Task> tasks) {
+        if (tasks.isEmpty()) {
+            System.out.println("No tasks in the list.");
+            return;
+        }
+
+        int count = 1;
+        System.out.println("Here are the tasks in your list:");
+        for (Task task: tasks) {
+            System.out.println(count + "." + task);
+            count += 1;
+        }
+    }
+    public static void printTask(Task task) {
+        System.out.println(task);
+    }
+    public static void printAddTask() {
```

Comment:
```suggestion
    public static void printAddedTask() {
```
try to make it more clear as to what the function does if not the reader might think that this function both prints and add